tests/MongoDB.Driver.Tests/Specifications/crud/CrudTestRunner.cs

```diff
@@ -19,55 +19,228 @@
 using System.IO;
 using System.Linq;
 using System.Reflection;
+using System.Threading;
 using MongoDB.Bson;
-using MongoDB.Bson.TestHelpers.XunitExtensions;
+using MongoDB.Driver.Core;
+using MongoDB.Driver.Core.Events;
+using MongoDB.Driver.Core.TestHelpers.JsonDrivenTests;
 using MongoDB.Driver.Core.TestHelpers.XunitExtensions;
+using MongoDB.Driver.TestHelpers;
 using Xunit;
 
 namespace MongoDB.Driver.Tests.Specifications.crud
 {
     public class CrudTestRunner
     {
+        #region static
+        private static readonly HashSet<string> __commandsToNotCapture = new HashSet<string>
+        {
+            "configureFailPoint",
+            "isMaster",
+            "buildInfo",
+            "getLastError",
+            "authenticate",
+            "saslStart",
+            "saslContinue",
+            "getnonce",
+            "find"
```

Comment:
I added `find` here to correspond to expected events from tests. Otherwise, we will have more events than expected in the spec tests.

Reply:
Sorry, I'm having trouble figuring this out: where are the additional `find` commands coming from?

Reply:
I didn't check it in debugging, but I suppose additional command is related to the fact that aggregation $out executes two operations. `Aggregate` (which is expected), and `find` (which is not expected from spec team): https://github.com/mongodb/mongo-csharp-driver/blob/7cd604ab4fee1c3a02fd56e7b3f99215055785f4/src/MongoDB.Driver.Legacy/MongoCollection.cs#L163

Reply:
Ah I see, so this only happens in Legacy?

Reply:
Ah, I see it happens in the non-Legacy driver as well (https://github.com/DmitryLukyanov/mongo-csharp-driver/blob/csharp2528/src/MongoDB.Driver/MongoCollectionImpl.cs#L112): after some thought, this does make sense to me :) Thanks for your patience!

Reply:
Yep :)